docs/learn/migrate/solana.mdx

```diff
@@ -1,8 +1,689 @@
+
+:::
+
+The release profile configuration is crucial for keeping your contract size under the 64KB limit:
+
+```toml
+[profile.release]
+opt-level = "z"
+overflow-checks = true
+debug = 0
+strip = "symbols"
+debug-assertions = false
+panic = "abort"
+codegen-units = 1
+lto = true
+
+[profile.release.build-override]
+opt-level = 3
+incremental = false
+codegen-units = 1
+```
+
+### Step 2: Convert Program Entry Point
+
+The biggest change you'll notice is how functions are defined and called.
+
+**Before (Solana):**
+
+```rust
+use solana_program::{
+    account_info::{next_account_info, AccountInfo},
+    entrypoint,
+    entrypoint::ProgramResult,
+    msg,
+    program_error::ProgramError,
+    pubkey::Pubkey,
+};
+
+entrypoint!(process);
+
+pub fn process(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    let accounts_iter = &mut accounts.iter();
+    let payer = next_account_info(accounts_iter)?;
+
+    msg!("Hello from Solana!");
+    Ok(())
+}
+```
+
+**After (Stellar):**
+
+```rust
+use soroban_sdk::{
+    contract, contractimpl, symbol_short, vec, Address, Env, Symbol, Vec,
+};
+
+#[contract]
+pub struct HelloContract;
+
+#[contractimpl]
+impl HelloContract {
+    pub fn hello(env: Env, to: Symbol) -> Vec<Symbol> {
+        vec![&env, symbol_short!("Hello"), to]
+    }
+}
+```
+
+The Stellar version is much more intuitive - just define your struct, implement the functions you want, and the framework handles the rest.
+
+### Step 3: Convert Account Handling
+
+Solana requires you to manually manage accounts and validate ownership. Stellar simplifies this significantly.
+
+**Before (Solana):**
+
+```rust
+pub fn process(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    let accounts_iter = &mut accounts.iter();
+    let user_account = next_account_info(accounts_iter)?;
+    let data_account = next_account_info(accounts_iter)?;
+
+    // Validate account ownership
+    if user_account.owner != program_id {
+        return Err(ProgramError::IncorrectProgramId);
+    }
+
+    // Access account data
+    let mut data = data_account.data.borrow_mut();
+    // ... process data
+    Ok(())
+}
+```
+
+**After (Stellar):**
+
+```rust
+use soroban_sdk::{
+    contract, contractimpl, symbol_short, vec, Address, Env, Symbol, Vec,
+};
+
+#[contract]
+pub struct DataContract;
+
+#[contractimpl]
+impl DataContract {
+    pub fn store_data(env: Env, user: Address, data: Symbol) {
+        // Store data with user as key
+        env.storage().instance().set(&user, &data);
+    }
+
+    pub fn get_data(env: Env, user: Address) -> Option<Symbol> {
+        env.storage().instance().get(&user)
+    }
+}
+```
+
+No more manual account validation or data borrowing - Stellar handles the storage abstraction for you.
```

Comment:
This example is really problematic, as it doesn't perform authorization (via `user.require_auth()`), and also makes a misleading statement that Stellar somehow handles storage authorization for you, which is not the case. This and the following examples should be updated to use proper authorization patterns both on Solana and Stellar, as these are critical for every smart contract.